llvm-general/src/LLVM/General/Internal/Attribute.hs

```diff
@@ -58,7 +65,13 @@ instance Monad m => EncodeM m A.FA.FunctionAttribute (Ptr FFI.FunctionAttrBuilde
     liftIO $ FFI.attrBuilderAddStringAttribute b kindP kindLen valueP valueLen
   encodeM a = return $ \b -> liftIO $ case a of
     A.FA.StackAlignment v -> FFI.attrBuilderAddStackAlignment b v
+    A.FA.AllocSize x y -> case y of
+                            Nothing -> FFI.attrBuilderAddAllocSize b x 0 0
+                            Just y' -> FFI.attrBuilderAddAllocSize b x 1 y'
```

Comment:
Better to avoid needing this kind of case and the magic use of 0/1 in the third argument.

Use currying/uncurrying helpers next to the FFI import (like constNamedStruct, for example) to bundle the flag and value together into a pair (or newtype with the others in LLVMCTypes like NothingAsMinusOne), and encode/decode instances in Coding.hs so that you can encode/decode y like everything else.